### ISSUE TITLE
Do we want to parametrize the type of optimizer?

### ISSUE THREAD

**Issue by jcohenadad** [ivadomed/ivadomed#1282]
### Motivation for the feature

By default, ivadomed's only uses Adam optimizer:
https://github.com/ivadomed/ivadomed/blob/fe4b6ad0a23db2b5b336b3e8b72cca9e5953bcd2/ivadomed/training.py#L134-L137

Do we want to expose other optimizers to the user via the config file?

**Comment by naga-karthik** [comment]
My thoughts: Everyone uses either Adam or [AdamW](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html#torch.optim.AdamW) these days. I wouldn't give much thought to which optimizer I’m using as long as it an adaptive optimizer (like Adam). My focus would rather be on tuning the learning rate that changing the optimizer itself.  

We could provide Adam, AdamW, RMSProp and SGD for the sake of giving more options but I'm confident that most users would stick to the Adam optimizer.
